### PR TITLE
Project-happy-thoughts-api- Marielle Ohlsson

### PR DESCRIPTION
**Pull request by Marielle8**
I aimed for the blue requirments int this project.

Link to deployed frontend on netlify: https://happy-thoughts8.netlify.app/

Link to deployed backend on heroku: https://marielles-happy-thoughts-api.herokuapp.com/thoughts

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,30 +1,81 @@
...
 
-// Start the server
-app.listen(port, () => {
-  // eslint-disable-next-line
-  console.log(`Server running on http://localhost:${port}`)
+//Endpoint that show thoughts with a limit on 20 thoughts per page
+app.get('/thoughts', async (_, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
+  res.json(thoughts)
 })
```
Comment (johlofg) [inline server.js:49]: Good comment and nice with _ instead of req.

**Review by johlofg — CHANGES_REQUESTED** [review]
Overall a good project with clean code and good comments. Everything look alright and it seems that the error handeling is working too. But I cant get the application to work?

File: server.js [hunk 2]
```diff
@@ -1,30 +1,81 @@
...
+// Defines the port the app will run on. 
+const port = process.env.PORT || 8082
 const app = express()
 
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlenght: 140
```
Comment (KarolCislowski) [inline server.js:22]: Nicely configured field 👍🏻

File: server.js [hunk 3]
```diff
@@ -1,30 +1,81 @@
...
+    }
+    res.status(400).json({ error: "Something went wrong" })
+  }
+})
+
+// Endpoint for likes
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const { thoughtId } = req.params
+  try {
+    const updatedThought = await Thought.findOneAndUpdate({ _id: thoughtId }, { $inc: { hearts: 1 } }, { new: true })
```
Comment (KarolCislowski) [inline server.js:69]: 💯

File: server.js [hunk 4]
```diff
@@ -1,30 +1,81 @@
...
+  res.send(listEndpoints(app))
 })
 
-// Start the server
-app.listen(port, () => {
-  // eslint-disable-next-line
-  console.log(`Server running on http://localhost:${port}`)
+// Endpoint that show thoughts with a limit on 20 thoughts per page
+app.get('/thoughts', async (_, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
```
Comment (KarolCislowski) [inline server.js:47]: 👍🏻

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Your code looks good, but your deployed server returns a 503 error
image
Check if you have correctly configured connection with mongo atlas

**Review by johlofg — APPROVED** [review]
Snyggt!

**Review by KarolCislowski — APPROVED** [review]
Great app 🥇

**Pull request closed without merging**